### ISSUE TITLE
Default to "Request power down" for VMs when quitting UTM app or closing a VM window

### ISSUE THREAD

**Issue by ireuben** [utmapp/UTM#5912]
Please browse the existing issues and filter by 'enhancement' to make sure this has not already been suggested!
It seems like it would be more friendly behavior if the default was to "Request power down" for a VM if you are quitting the app or closing a VM window, so it would shut down gracefully and not risk losing data. If that doesn't work, then the user could decide to force a shut down or kill.

I'm using Windows 11 and a command-line only linux, and they both shut down automatically after a "Request power down" (no further prompts need to be acknowledged). 

As a bonus, could have a timeout (30 seconds?) which then pops up a dialog saying VM doesn't seem to be responding to the request to shutdown, so would you like to force it and possibly lose data?

Could also see this as a pref ("When closing a running VM: [Request power down]", with the same three menu options as the power icon on VM window). But does seem like a default of "Request power down" would be ideal both for usability (it just works with some VMs) and data safety (won't mess up your VM's drive, and worst case you have to click Shutdown on the prompt inside VM).

**Comment by Mister-A** [comment]
As of 4.4.4 the default action is to suspend a running guest if the guest machine supports it - so it is now safe to close the VM's window in that case. As long as you have the client tools installed in the guest to keep the real-time clock synced it is now much more usable as a result.

However I do agree allowing us to set the default close action would be by far the best option.

Related issue: #4001

**Comment by ca0x** [comment]
Any update on this? 

Can we have the option on what closing the window does?

Since save state is not supported on some VMs, I would prefer to have the option of auto requesting power off, if possible.

**Comment by kenji21** [comment]
Maybe interesting : https://forums.developer.apple.com/forums/thread/702160

**Comment by kenji21** [comment]
Oh, I just discovered that we can long clic on the shut down button to "request to power down"

**Comment by epheterson** [comment]
I agree that the option to request shutdown would be ideal. This solution doesn't work with GPU acceleration:
`Suspend is not supported when GPU acceleration is enabled.`

> Oh, I just discovered that we can long clic on the shut down button to "request to power down"

"Request power down" should be the default action for that button! I didn't think it was supported

**Comment by ca0x** [comment]
@osy Thank you for your work on this! I was wondering if it might be possible to add this feature and include the default action as an option in the settings.

It would be really beneficial if shutting down the host could automatically shut down the VMs. Currently, shutting down or rebooting the host disrupts the VMs since suspend doesn’t seem to be supported with Apple Virtualization. I receive the following error: "An error occurred while restoring the virtual machine. The virtual machine failed to restore with error 'invalid argument.'"

Even if suspend worked correctly, having the option to choose would be much preferred, as many Docker services fail on resume after the VM has been suspended for a while.

Additionally, it would be fantastic to have an auto-start option for VMs as well.

Thank you for considering this!
